src/client/src/Client.c

```diff
@@ -395,12 +506,23 @@ STATUS stopKinesisVideoStreams(CLIENT_HANDLE clientHandle)
         }
     }
 
+    // unlock streams list lock after iteration
+    pKinesisVideoClient->clientCallbacks.unlockMutexFn(pKinesisVideoClient->clientCallbacks.customData,
+                                                       pKinesisVideoClient->base.streamListLock);
+    streamsListLock = FALSE;
+
+
 CleanUp:
 
     if (releaseClientSemaphore) {
         semaphoreRelease(pKinesisVideoClient->base.shutdownSemaphore);
     }
 
+    if (streamsListLock) {

```

Comment:
Move this just above the releaseClientSemaphore. Shouldn't matter really in this case but will be future code change proof